test/js/bun/shell/bunshell-default.test.ts

```diff
@@ -1,25 +1,54 @@
 import { $ } from "bun";
+import { bunExe, createTestBuilder } from "./test_builder";
+import { bunEnv } from "harness";
+const TestBuilder = createTestBuilder(import.meta.path);
 
 test("default throw on command failure", async () => {
-  try {
-    await $`echo hi; ls oogabooga`.quiet();
-    expect.unreachable();
-  } catch (e: any) {
-    expect(e).toBeInstanceOf(Error);
-    expect(e.exitCode).toBe(1);
-    expect(e.message).toBe("Failed with exit code 1");
-    expect(e.stdout.toString("utf-8")).toBe("hi\n");
-    expect(e.stderr.toString("utf-8")).toBe("ls: oogabooga: No such file or directory\n");
-  }
+  // Run in a subproc because other tests may change the value of $.throws
```

Comment:
A thought: Maybe it would make sense to allow users to create a shell instance that is configurable and not global, e.g. `const $ = Bun.$.withOptions(...);`. This would alleviate the issues here. (obviously this is out of scope for this PR)

Reply:
Yeah we should do this